### ISSUE TITLE
\n is replaced with \r\n in attachments

### ISSUE THREAD

**Issue by jakobmulvad** [nodemailer/nodemailer#815]
I'm updating from a deprecated version of nodemailer and suddenly I have test cases that are failing. It seems the latest nodemailer will alter the content of attachments if they are `text/plain`. Specifically `\n` is replaced with `\r\n`. This happens because the content is transferred using `content-transfer-encoding: quoted-printable`. This will cause the content to be affected by the general "lets replace all \n with \r\n" in https://github.com/nodemailer/nodemailer/blob/master/lib/smtp-connection/data-stream.js#L55. 

This is a clear bug in my opinion. When I provide a buffer to nodemailer as the contents of an attachment then I expect that buffer to delivered verbatim.

I can work around this by manually setting the `contentTransferEncoding` to `base64` but that shouldn't really be necessary. 

I have attached an isolated test case in the bottom.

Thanks for a great module!

### 1. Is this a bug in Nodemailer?
 
 - [X] Yes
 - [ ] No
 
> Note. Non bug issues are closed by default. Head to Stack Overflow for support questions: https://stackoverflow.com/search?q=nodemailer

### 2. Is this related to Gmail / Hotmail / Yahoo?

 - [ ] Yes
 - [x] No
 
 > Note. If you can not log in to Gmail then Gmail is blocking you. These issues are closed by default.

### 3. Is your issue related to SSL/TLS errors?

 - [ ] Yes
 - [x] No
 
 > Note. Most probably you are trying to connect to a server with a self-signed/expired cert or the server supports only ancient ciphers or you are using an antivirus that intercepts your internet connection. Such issues are closed by default.

### 4. Is your issue related to connection timeouts

 - [ ] Yes
 - [x] No

> Note. Most probably you are firewalled out from the server. Such issues are closed by default.

### 5. Do you get SyntaxError when using Nodemailer?

 - [ ] Yes
 - [x] No

> Note. You are using too old version of Node.js, please upgrade. Such issues are closed by default.

### 6. Nodemailer version you are having problems with (eg. v1.3.7)
v4.3.1

### 7. Node.js version you are using (run `node -v` to see it, eg v5.5.0)
v8.8.1

### 8. Your operating system (eg. Windows 10, Ubuntu 14.04 etc.)
Windows 10

### 9. Nodemailer configuration settings (if applicable)
??

### 10. If possible, include a minimal test case that can be used to verify your issue (link to a gist would be great!)
```
const assert = require('assert')
const { createTransport } = require('nodemailer')
const { SMTPServer } = require('smtp-server')
const { simpleParser } = require('mailparser')

const content = new Buffer('Hello World!\n')
const contentBase64 = content.toString('base64')

const smtpServer = new SMTPServer({
	authOptional: true,
	onData(stream, session, callback) {
		simpleParser(stream, (err, email) => {
			console.log(email.attachments[0].content, content)
			assert.equal(email.attachments[0].content.toString('base64'), contentBase64, 'Attachment contents should en un-altered')
			callback()
			smtpServer.close()
		})
	},
}).listen(587, () => {
	createTransport({
		host: 'localhost',
		port: 587,
		tls: { rejectUnauthorized: false },
	}).sendMail({
		to: 'test@test.com',
		attachments: [{
			filename: 'text.txt',
			contentType: 'text/plain',
			content,
		}]
	})
})
```

**Comment by andris9** [comment]
This is intentional. \n line breaks are not allowed in rfc822 messages, Nodemailer normalizes all linebreaks to \r\n. If you want to use \n, then use base64 encoding.

**Comment by andris9** [comment]
Here's the relevant RFC rule:

```
2.3.  Body

   The body of a message is simply lines of US-ASCII characters.  The
   only two limitations on the body are as follows:

   o  CR and LF MUST only occur together as CRLF; they MUST NOT appear
      independently in the body.
```

**Comment by jakobmulvad** [comment]
I understand that, but wouldn't you agree sending attachments using `content-transfer-encoding: quoted-printable` per default is incorrect in that case? You are altering the file contents because of a technical detail in the underlying protocol.

Consider situations where file content is checksummed on delivery and on arrival - that would fail. Essentially the file you are sending is not the same file you are receiving.

Defaulting to `content-transfer-encoding: base64` for all attachments seems better in almost all cases. If you are very concerned about the transfer size or if you absolutely want to see the content in plain text in the smtp body then you can always manually use `quoted-printable`.

**Comment by andris9** [comment]
Actually, now that I think of it, then you're right. Using \r\n for line breaks is correct but defaulting to 7bit transfer encoding for text attachments is not as 7bit can not guarantee integrity during transmission. It isn't just about line breaks, also empty lines could be added or removed from the end / beginning of the content.

I changed the default behavior to use base64 for attachments and released the change as v4.4.0

**Comment by robert-claypool** [comment]
Minor point: this rule is in https://tools.ietf.org/html/rfc2822#section-2.3 (not the older rfc822).
